generic/tclProc.c

```diff
@@ -1513,6 +1536,337 @@ InitArgsAndLocals(
     return ProcWrongNumArgs(interp, skip);
 }
 
+/*
+ *----------------------------------------------------------------------
+ *
+ * TclNamedParametersRewrite
+ *
+ * Change the call parameters to match the "named" parameters of a proc.
+ * Using the variable names of the parameters, create the argument list
+ * by converting -name value pairs into positional elements in list.
+ *
+ * Named parameters are respented as "-key value" pairs where key is the proc
+ * parameter variable. The general form is zero or more key value parameters.
+ * Arguments are processed from left to right.  If the argument starts with -,
+ * then this is a key and will be followed by a value.
+ * An optional "--" will end the key value processing and the remaining arguments will
+ * be positional.
+ *
+ *     {-key value}* [--] {positional}*
+ *
+ *----------------------------------------------------------------------
+ */
+
+static int TclNamedParametersRewrite(Tcl_Interp *interp, CallFrame *framePtr) {
+    Interp *iPtr = (Interp *) interp;
+    int srcIndex = 1, npos = 0, varArgs = 0, markerIndex = 0;
+    Tcl_Obj *copyPtr = NULL;
+
+    if (!framePtr || !framePtr->procPtr || !framePtr->procPtr->firstLocalPtr) {
+        return TCL_ERROR;
+    }
+
+    /* Find where the -- is located in the proc declaration.
+     * And determine if the proc has variable number of arguments.
+     * Caching this information when compiling the proc would be faster. */
```

Comment:
So every time any proc is called this code will loop through the arguments defined in the proc to look for --?